budget app backend/server.js

```diff
@@ -1,24 +1,34 @@
+const { PORT, JWT_SECRET } = require("./config");
 const express = require("express");
 const app = express()
 const fs = require("fs")
 const path = require("path");
-// I DID NOT WANT TO USE passport.js because its configuration was terribly confusing. We could just use cookie to store jwt. By the way it is the same as storing "Bearer jwt_token" in "Authorization" header. No complicated stuff like JWTStrategy. passport.js maybe useful with other authentication methods like google/instagram auth. Do you understand what I mean ? I cannot digest why we are using passport stuff. I understand what Cookie is and I don't think it's a frontend stuff (as many other dev's are saying). Actually it's a backend stuff. I'm unhappy with the fact that we are blindly learning (looking through) ready-made libraries without knowing what is actually happening or why "this_package" was used "here"
-
-const { PORT } = require("./config");
+const passport = require("passport")
+const jwtStrategy = require("./modules/passport-jwt")
+// I DID NOT WANT TO USE passport.js because its configuration was terribly confusing. We could just use cookie to store jwt. By the way it is the same as storing "Bearer jwt_token" in "Authorization" header. No complicated stuff likejwtTStrategy. passport.js maybe useful with other authentication methods like google/instagram auth. Do you understand what I mean ? I cannot digest why we are using passport stuff. I understand what Cookie is and I don't think it's a frontend stuff (as many other dev's are saying). Actually it's a backend stuff. I'm unhappy with the fact that we are blindly learning (looking through) ready-made libraries without knowing what is actually happening or why "this_package" was used "here"
 
 // middlewares
 app.use(express.json()); //to parse body's json
 app.use(express.urlencoded({ extended: true }))
-
-
+app.use(passport.initialize())
 // const cookie_parser = require("cookie-parser")
 // app.use(cookie_parser()) //to work with jwt in Cookie
 
+passport.use(jwtStrategy)
+let auth = passport.authenticate("jwt", {session: false})
+// how to use this middleware in specific route inside routes folder ?
+
 // importing all routers
 fs.readdir(path.join(__dirname, "routes"), (err, files) => {
     files.forEach(file => {
         const {endpoint, router} = require(path.join(__dirname, "routes", file))
-        if(endpoint && router) app.use(endpoint, router)
+        if(endpoint && router) {
+            if(endpoint !== "/auth"){
```

Comment:
I couldn't extract auth middleware (line 18) to use only in specific routes. Because, passport should be called in app.js (line 13). Is there any way to export auth middleware so that I could use it in, for example, income_route.js ?

Reply:
I dont understand why you use fs.readdir to import routes? Why dont you like simple import?

Reply:
> I dont understand why you use fs.readdir to import routes? Why dont you like simple import?

I used fs because it dynamically imports all route.js files in routes folder. If I create a new route.js file, I don't have to manually import/export.